homeassistant/components/vesync/humidifier.py

```diff
@@ -0,0 +1,201 @@
+    def discover(devices):
+        """Add new devices to platform."""
+        _setup_entities(devices, async_add_entities, coordinator)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(hass, VS_DISCOVERY.format(VS_HUMIDIFIERS), discover)
+    )
+
+    _setup_entities(hass.data[DOMAIN][VS_HUMIDIFIERS], async_add_entities, coordinator)
+
+
+@callback
+def _setup_entities(
+    devices: list[VeSyncBaseDevice],
+    async_add_entities,
+    coordinator: VeSyncDataCoordinator,
+):
+    """Add humidifier entities."""
+    entities = [
+        VeSyncHumidifierHA(dev, coordinator) for dev in devices if is_humidifier(dev)
+    ]
+
+    async_add_entities(entities, update_before_add=True)
+
+
+def _get_ha_mode(vs_mode: str) -> str | None:
+    ha_mode = VS_TO_HA_MODE_MAP.get(vs_mode)
+    if ha_mode is None:
+        _LOGGER.warning("Unknown mode '%s'", vs_mode)
+    return ha_mode
+
+
+def _get_vs_mode(ha_mode: str) -> str | None:
+    vs_mode = HA_TO_VS_MODE_MAP.get(ha_mode)
+    if vs_mode is None:
+        _LOGGER.warning("Unknown mode '%s'", ha_mode)
+    return vs_mode
+
+
+class VeSyncHumidifierHA(VeSyncBaseEntity, HumidifierEntity):
+    """Representation of a VeSync humidifier."""
+
+    # The base VeSyncBaseEntity has _attr_has_entity_name and this is to prevent
+    # none being appended to entity id.
+    _attr_name = ""
+
+    _attr_max_humidity = MAX_HUMIDITY
+    _attr_min_humidity = MIN_HUMIDITY
+    _attr_supported_features: HumidifierEntityFeature = HumidifierEntityFeature(
+        HumidifierEntityFeature.MODES
+    )
+
+    def __init__(
+        self,
+        humidifier: VeSyncHumidifierDevice,
+        coordinator: VeSyncDataCoordinator,
+    ) -> None:
+        """Initialize the VeSync humidifier device."""
+        super().__init__(humidifier, coordinator)
+        self.smarthumidifier = humidifier
+
+    @property
+    def available_modes(self) -> list[str]:
+        """Return the available mist modes."""
+        return [
+            ha_mode
+            for ha_mode in (
+                _get_ha_mode(vs_mode) for vs_mode in self.smarthumidifier.mist_modes
+            )
+            if ha_mode
+        ]
+
+    @property
+    def target_humidity(self) -> int:
+        """Return the humidity we try to reach."""
+        return self.smarthumidifier.config["auto_target_humidity"]
+
+    @property
+    def mode(self) -> str | None:
+        """Get the current preset mode."""
+        return _get_ha_mode(self.smarthumidifier.details["mode"])
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any]:
+        """Return the state attributes of the humidifier."""
+
+        attr = {}
+        for k, v in self.smarthumidifier.details.items():
+            if k in VS_TO_HA_ATTRIBUTES:
+                attr[VS_TO_HA_ATTRIBUTES[k]] = v
+            elif k in self.state_attributes:
+                attr[f"vs_{k}"] = v
+            else:
+                attr[k] = v
+        return attr
```

Comment:
Let's not add extra states attributes. If something is interesting enough we can make a separate sensor for it